packages/realm/package.json

```diff
@@ -74,72 +74,64 @@
         "build:node"
       ]
     },
-    "build": {
-      "dependencies": [
-        "bundle",
-        "build:node",
-        "build:android",
-        "build:ios"
-      ]
-    },
     "bundle": {
       "command": "rollup --config",
       "dependencies": [
-        "../realm-network-transport:build",
-        "build:bindgen:typescript",
-        "build:bindgen:node-wrapper"
+        "../realm-network-transport:bundle",
+        "bindgen:generate:typescript",
+        "bindgen:generate:wrappers"
       ],
       "files": [
         "rollup.config.mjs",
         "src/**/*.ts",
         "tsconfig.json",
         "src/**/tsconfig.json"
       ],
-      "output": [
-        "dist/bundle.*.js",
-        "dist/bundle.*.mjs"
-      ]
+      "output": ["dist"]
     },
-    "build:bindgen": {
-      "command": "cmake-js build --debug --directory ../bindgen",
+    "bindgen:configure": {
+      "command": "cmake-js configure --debug --directory ../bindgen",
+      "dependencies": ["../..:submodules"]
+    },
+    "bindgen:build:node": {
+      "command": "cmake --build ../bindgen/build --target realm-js-node",
+      "dependencies": ["bindgen:configure"]
+    },
+    "bindgen:generate:typescript": {
+      "command": "realm-bindgen --template typescript --spec ../bindgen/spec.yml --spec ../bindgen/js_spec.yml --output ./generated/ts",
+      "dependencies": [
+        "../..:submodules",
+        "../bindgen:generate:spec-schema"
+      ],
       "files": [
         "../bindgen/spec.yml",
-        "../bindgen/src/**/*.ts",
-        "../bindgen/src/**/*.h"
-      ],
-      "dependencies": [
-        "../..:submodules"
+        "../bindgen/js_spec.yml",
+        "../bindgen/src"
       ],
       "output": [
         "generated/ts/core.ts",
-        "generated/ts/native.d.mts",
-        "generated/ts/native.mjs",
-        "generated/ts/native-rn.mjs"
+        "generated/ts/native.d.mts"
       ]
     },
-    "build:bindgen:typescript": {
-      "command": "realm-bindgen --spec ../bindgen/spec.yml --spec ../bindgen/js_spec.yml --template typescript --output ./generated/ts",
+    "bindgen:generate:wrappers": {
+      "command": "realm-bindgen --template node-wrapper --spec ../bindgen/spec.yml --spec ../bindgen/js_spec.yml --output ./generated/ts",
       "dependencies": [
         "../..:submodules",
         "../bindgen:generate:spec-schema"
-      ]
-    },
-    "build:bindgen:node-wrapper": {
-      "command": "realm-bindgen --spec ../bindgen/spec.yml --spec ../bindgen/js_spec.yml --template node-wrapper --output ./generated/ts",
-      "dependencies": [
-        "../..:submodules",
-        "../bindgen:generate:spec-schema"
-      ]
-    },
-    "rebuild:bindgen": {
-      "command": "cmake-js rebuild --debug --directory ../bindgen",
-      "dependencies": [
-        "../..:submodules"
+      ],
+      "files": [
+        "../bindgen/spec.yml",
+        "../bindgen/js_spec.yml",
+        "../bindgen/src"
```

Comment:
Consider excluding `src/templates` and then re-including the specific template used by the command. That matches the cmake behavior and avoids rerunning one generator when you only modify another.

Ditto for all similar tasks.